### DISCUSSION TITLE
CLI doing nothing

### DISCUSSION THREAD

**Discussion by bc3tech** [shlinkio/shlink#1973] [Help wanted] [answered]
### Shlink version

3.7.3

### PHP version

8.3

### How do you serve Shlink

Self-hosted Apache

### Database engine

MySQL

### Database version

5.7.23-23

### Summary

I've got shlink up and running, the app.shlink.io works perfectly and redirects are working just fine.

My next task is to import the URLs from my old Yourls instance, so I turned to the cli.
When I run it in the terminal, however, it does nothing:

![image](https://github.com/shlinkio/shlink/assets/1967433/16d47ce3-2877-4a18-832a-1a39e6834342)

As shown in that screenshot as well, it's marked as executable on disk, so I'm not sure what might be happening. Any idea?
The odd thing is I had installed this previously (debugging) and was able to use CLI w/o issue, but in this latest installation I'm not (and I don't recall doing anything differently)

I noticed an `error_log` file in the `bin/` folder, so I deleted it and ran `./cli -h` again, and it appeared again.

The content of it is

```txt
[10-Jan-2024 12:29:45 America/Chicago] PHP Parse error:  syntax error, unexpected '->' (T_OBJECT_OPERATOR), expecting ')' in /home2/bc3tech/public_html/.website_40dd84f1/config/container.php on line 27
```

Inspecting this line I find:

```php
date_default_timezone_set(EnvVars::TIMEZONE->loadFromEnv(date_default_timezone_get()));
```

which makes me think the Time Zone isn't being inferred from the environment, and I should've set it during installation.

Is there a way I can set this now?

**Comment by acelaya** [comment, marked as answer]
That error means the CLI is being run with the wrong PHP version, one that is not supported by Shlink.

Latest Shlink version requires PHP >=8.2

**Reply by bc3tech** [reply]
hm. when i configured the installation i told it 8.3 and that's what it says it's running on my hosting provider... i'll dig around, thanks.

**Reply by acelaya** [reply]
Try running `php ./cli` instead. The script references to what is usually the default location for the PHP binary, but perhaps that's an older version than the one in your path.

Ultimately, prefixing it with the absolute path to a binary you know it's new enough, should also work.

**Reply by bc3tech** [reply]
hm. ok so the site is running on 8.3 because it works just fine. But the CLI, via terminal, is picking up php 7.4 (default on my provider):

```sh
# php --version
PHP 7.4.33 (cli) (built: Dec 19 2023 14:59:00) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with the ionCube PHP Loader + ionCube24 v10.4.5, Copyright (c) 2002-2020, by ionCube Ltd.
    with Zend OPcache v7.4.33, Copyright (c), by Zend Technologies
```

i wonder how i can get it to pick up the 8.3 executable... i'll dig around.

**Reply by bc3tech** [reply]
Got it, they have multiple installs and the default `php` one is 7.4
to get 8.3 i have to do `ea-php83 ./cli -h` and then everything works.

thanks for the php version tip!
